cli/src/semgrep/rule_match.py

```diff
@@ -356,6 +356,18 @@ def get_end_line_hash(self) -> str:
         """
         last_line = self.get_individual_line(self.end.line)
         return hashlib.sha256(last_line.encode()).hexdigest()
+    
+    def is_direct(self) -> bool:
+        return (
+            self.extra["sca_info"].dependency_match.found_dependency.transitivity
+            == "direct"
+        )
+    
+    def is_reachable_or_always_reachable(self):
```

Comment:
Would also rename this to `is_reachable_or_always_reachable_sca_rule` and add the check that `"sca_info"` is present.